components/core/src/Grep.cpp

```diff
@@ -497,48 +497,51 @@ std::optional<Query> Grep::process_raw_query(
         log_surgeon::lexers::ByteLexer& reverse_lexer,
         bool use_heuristic
 ) {
-    Query query;
-    // Set properties which require no processing
-    query.set_search_begin_timestamp(search_begin_ts);
-    query.set_search_end_timestamp(search_end_ts);
-    query.set_ignore_case(ignore_case);
-
     // Add prefix and suffix '*' to make the search a sub-string match
     string processed_search_string = "*";
     processed_search_string += search_string;
     processed_search_string += '*';
-
-    // Clean-up search string
     processed_search_string = clean_up_wildcard_search_string(processed_search_string);
-    query.set_search_string(processed_search_string);
 
     // Split search_string into tokens with wildcards
     vector<QueryToken> query_tokens;
     size_t begin_pos = 0;
     size_t end_pos = 0;
     bool is_var;
+    string search_string_for_sub_queries{processed_search_string};
     if (use_heuristic) {
         // Replace non-greedy wildcards with greedy wildcards since we currently have no support for
-        // searching compressed files with non-greedy wildcards
-        std::replace(processed_search_string.begin(), processed_search_string.end(), '?', '*');
+        // generating subqueries with '?' wildcards. The wildcard match on the decompressed message
+        // mitigates this limitation.
```

Comment:
Not sure if mitigate is the right word here, doesn't the wildcard match on the decompressed message fully ignore this limitation

Reply:
Good point.